src/codeExecutionWorker.js

```diff
@@ -0,0 +1,67 @@
+/* global eval */
+'use strict'
+import { connectIframe, PluginClient } from '@remixproject/plugin'
+import ethers from 'ethers' // eslint-disable-line
+import Web3 from 'web3' // eslint-disable-line
+import swarmgw_fn from 'swarmgw' // eslint-disable-line
+window.swarmgw = swarmgw_fn() // eslint-disable-line
+window.ethers = ethers
+window.Web3 = Web3
+
+class Web3Provider {
```

Comment:
I'm not sure why you need a class for that. You're not using inheritance not instanciating it several times.
You can do : 
```javascript
const client = new CodeExecutor()
const provider = {
  sendAsync(payload, callback) {
    client.call('appWeb3Provider', 'sendAsync', payload)
      .then(result => callback(null, result)
      .catch(e => callback(e))
  }
}
```

Also I thought the Web3 provider exposed a Promise interface now instead of a callback one...